scripts/bert/fp16_utils.py

```diff
@@ -20,6 +20,7 @@
 import collections
 import mxnet as mx
 from mxnet import nd
+from collections import defaultdict
 
 def grad_global_norm(parameters, max_norm):
     """Calculate the 2-norm of gradients of parameters, and how much they should be scaled down
```

Comment:
Actually, would you mind helping move this function to src/gluonnlp/utils/parameter.py?

API Spec:
```
def grad_global_norm(parameters, max_norm=None):
    """Calculate the 2-norm of gradients of parameters, and how much they should be scaled down
    such that their 2-norm does not exceed `max_norm`, if `max_norm` if provided.

    If gradients exist for more than one context for a parameter, user needs to explicitly call
    ``trainer.allreduce_grads`` so that the gradients are summed first before calculating
    the 2-norm.

    .. note::
        This function is only for use when `update_on_kvstore` is set to False in trainer.

    Example::

        trainer = Trainer(net.collect_params(), update_on_kvstore=False, ...)
        for x, y in mx.gluon.utils.split_and_load(X, [mx.gpu(0), mx.gpu(1)]):
            with mx.autograd.record():
                y = net(x)
                loss = loss_fn(y, label)
            loss.backward()
        trainer.allreduce_grads()
        norm = grad_global_norm(net.collect_params().values())
        ...

    Parameters
    ----------
    parameters : list of Parameters
    max_norm: NDArray, optional
        The maximum L2 norm threshold. If provided, `ratio` and `is_finite` will be returned.

    Returns
    -------
    NDArray
      Total norm. Shape is (1,)
    NDArray
      Ratio for rescaling gradients based on max_norm s.t. grad = grad / ratio.
      If total norm is NaN, ratio will be NaN, too.
      Returned if `max_norm` is provided. Shape is (1,)
    NDArray
      Whether the total norm is finite, returned if `max_norm` is provided. Shape is (1,)
    """

```

Then we can add an unit test for `gluonnlp.utils.grad_global_norm`. We can update the implementation of `gluonnlp.utils.clip_grad_global_norm` to call `grad_global_norm` first, too.